digiroad2-oracle/src/main/scala/fi/liikennevirasto/digiroad2/dao/RoadLinkDAO.scala

```diff
@@ -527,14 +527,37 @@ class RoadLinkDAO {
     }
   }
 
+  def getLatestModification(modifications: Seq[Modification]): Modification = {
+    def calculateLatestModification(a: (DateTime, String), b: (DateTime, String)): (DateTime, String) = {
+      if (a._1.isAfter(b._1)) a else b
+    }
+
+    val mappedModifications = modifications.flatMap {
+      case (Modification(Some(at), Some(by))) => Some((at, by))
+      case _ => None
+    }
+    mappedModifications.reduceOption(calculateLatestModification) match {
+      case Some((date, by)) => Modification(Some(date), Some(by))
+      case None => Modification(None, Some(""))
+    }
+  }
+
   def extractPrivateRoadInfo(r: PositionedResult): Map[String, Any] = {
     val privateRoadInfoArray = r.nextObjectOption()
+    def cleanValue(value: String): String = {
+      value.stripPrefix("\"").stripSuffix("\"")
+    }
     val privateRoadInfoSeq: Seq[(Option[String], Option[String], Option[String], Option[String])] = privateRoadInfoArray match {
       case Some(sqlArray: SqlArray) =>
         sqlArray.getArray.asInstanceOf[Array[AnyRef]].collect {
           case pgObject: PGobject =>
             val Array(name, value, modifiedDate, modifiedBy) = pgObject.getValue.stripPrefix("(").stripSuffix(")").split(",", -1)
-            (Option(name).filter(_.nonEmpty), Option(value).filter(_.nonEmpty), Option(modifiedDate).filter(_.nonEmpty), Option(modifiedBy).filter(_.nonEmpty))
+            (
```

Comment:
Mistä nämä ylimääräiset lainaukset tulee stringiin? Onko kannassa ylimääräiset lainausmerkit arvojen ympärillä vai tuleeko nämä arvojen haun yhteydessä?

Reply:
Ylimääräiset hipsut taitaa tosiaan olla seurausta tavasta, jolla aggregaatti yhdistää yksityistietiedot yhdeksi sarakearvoksi sql:ssä. Varmistan vielä.

Reply:
Sanoisin tosiaan, että ylimääräiset hipsut syntyvät juurikin siitä, kun tällä funktiolla luotu sql-array muutetaan ja puretaan Scala-objektiksi:

`ARRAY_AGG(ROW(rla.name, rla.value, COALESCE(rla.modified_date, rla.created_date), COALESCE(rla.modified_by, rla.created_by)))
        FILTER (WHERE rla.name IS NOT NULL) AS private_road_info`
        
Arvojen ympärille syntyy tuossa "".        
        
Sanoisin että ylimääräisten hipsujen siivoaminen on tässä tapauksessa luettavuuden kannalta helpoin toteutus.